### PR TITLE
[contract_indexer] fix transfer event handling

### PR DESCRIPTION
**Pull request by envestcc**
# Description
Currently, only the transfer events triggered during the creation of a bucket are being handled. The transfer events triggered during subsequent bucket transfers are not being handled correctly.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: blockindex/contractstaking/event_handler.go [hunk 1]
```diff
@@ -434,7 +434,15 @@ func (eh *contractStakingEventHandler) handleTransferEvent(event eventParam) err
 		return err
 	}
 
-	eh.tokenOwner[tokenID.Uint64()] = to
```
Comment (Liuhaai) [inline blockindex/contractstaking/event_handler.go:437]: same issue on L507?

File: blockindex/contractstaking/event_handler.go [hunk 2]
```diff
@@ -434,7 +434,15 @@ func (eh *contractStakingEventHandler) handleTransferEvent(event eventParam) err
...
+	token := tokenID.Uint64()
```
Comment (Liuhaai) [inline blockindex/contractstaking/event_handler.go:437]: `token` -> `tokenID`

File: blockindex/contractstaking/event_handler.go (hunk 1 above)
Comment (dustinxie) [reply blockindex/contractstaking/event_handler.go:437]: yes, please do a complete revisit of all the handlers, see if there are other places need to cover

File: e2etest/contract_staking_test.go [hunk 3]
```diff
@@ -1715,6 +1715,62 @@ func TestContractStaking(t *testing.T) {
 		r.True(ok)
 		r.EqualValues(identityset.Address(delegateIdx).String(), bt.Candidate.String())
 	})
+
+	t.Run("transfer token", func(t *testing.T) {
```
Comment (dustinxie) [inline e2etest/contract_staking_test.go:1719]: if this test case is added in first place, it should've caught this bug, is that right?
like we discussed, need to add comprehensive tests, not to cover every line of code, but all newly added funcs/APIs should be covered

File: blockindex/contractstaking/event_handler.go (hunk 1 above)
Comment (envestcc) [reply blockindex/contractstaking/event_handler.go:437]: only `handleTransferEvent` need to update

File: blockindex/contractstaking/event_handler.go (hunk 2 above)
Comment (envestcc) [reply blockindex/contractstaking/event_handler.go]: fixed

**Review by millken — APPROVED** [review]

**Review by dustinxie — APPROVED** [review]

**Pull request merged**